frame/pallet-account/src/pallet/mod.rs

```diff
@@ -0,0 +1,228 @@
+        traits::IsSubType,
+    };
+    use frame_system::pallet_prelude::*;
+    use sp_runtime::traits::{IdentifyAccount, Verify};
+
+    /// The current storage version.                                                                                      
+    const STORAGE_VERSION: StorageVersion = StorageVersion::new(1);
+
+    #[pallet::pallet]
+    #[pallet::generate_store(pub(super) trait Store)]
+    #[pallet::storage_version(STORAGE_VERSION)]
+    #[pallet::without_storage_info]
+    pub struct Pallet<T>(PhantomData<T>);
+
+    #[pallet::config]
+    pub trait Config: frame_system::Config {
+        /// Custom origin type.
+        type CustomOrigin: Parameter + TryInto<Self::AccountId>;
+        /// Parameter that defin different origin options and how to create it.
+        type CustomOriginKind: Parameter + OriginDeriving<Self::AccountId, Self::CustomOrigin>;
+        /// The runtime origin type.
+        type RuntimeOrigin: From<Self::CustomOrigin>
+            + From<frame_system::RawOrigin<Self::AccountId>>;
+        /// The overarching call type.
+        type RuntimeCall: Parameter
+            + Dispatchable<RuntimeOrigin = <Self as Config>::RuntimeOrigin>
+            + GetDispatchInfo
+            + From<frame_system::Call<Self>>
+            + IsSubType<Call<Self>>
+            + IsType<<Self as frame_system::Config>::RuntimeCall>;
+        /// Meta transaction chain magic prefix. Required to prevent tx replay attack.
+        type ChainMagic: Get<u16>;
+        /// Meta transaction signature type.
+        type Signature: Parameter + Verify<Signer = Self::Signer>;
+        /// Meta transaction signer type.
+        type Signer: IdentifyAccount<AccountId = Self::AccountId>;
+        /// General event type.
+        type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+        /// Weight information for extrinsics in this pallet.
+        type WeightInfo: WeightInfo;
+    }
+
+    #[pallet::error]
+    pub enum Error<T> {
+        /// Origin with given index not registered.
+        UnregisteredOrigin,
+        /// Signer not match signature, check nonce, magic and try again.
+        BadSignature,
+    }
+
+    #[pallet::event]
+    #[pallet::generate_deposit(pub(crate) fn deposit_event)]
+    pub enum Event<T: Config> {
+        NewOrigin {
+            account: T::AccountId,
+            origin: T::CustomOrigin,
+        },
+        ProxyCall {
+            payer: T::AccountId,
+            origin: T::CustomOrigin,
+            result: DispatchResult,
+        },
+        MetaCall {
+            payer: T::AccountId,
+            origin: T::AccountId,
+            result: DispatchResult,
+        },
+    }
+
+    #[pallet::origin]
+    pub type Origin<T> = <T as Config>::CustomOrigin;
+
+    /// Account origins
+    #[pallet::storage]
+    pub type AccountOrigin<T: Config> =
+        StorageMap<_, Blake2_128Concat, T::AccountId, Vec<T::CustomOrigin>, ValueQuery>;
```

Comment:
`Vec<_>` should not be used in storage. Instead, you should use `BoundedVec`. That way you can also get rid of the `without_storage_info` macro you used above. This item cannot be _unlimited_.

Reply:
Used double-map instead of vector.